sklearn/ensemble/partial_dependence.py

```diff
@@ -306,8 +306,8 @@ def convert_feature(fx):
                 l.append(feature_names[i])
             names.append(l)
     except IndexError:
-        raise ValueError('features[i] must be in [0, n_features) '
-                         'but was %d' % i)
+        raise ValueError('features[i] must be in [0, len(feature_names)={0} ) '
+                         'but was {1}'.format(len(feature_names), i))
```

Comment:
There is a bit of history here that's missing, because I picked this up from an abandoned PR. The error message that I have here is, more or less, the same as that proposed in the previous PR, #7609 .